core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -110,21 +110,32 @@ public void commitState(Snapshot snapshot)
       throws CommitException, UnknownTransactionStatusException {
     String id = snapshot.getId();
     try {
-      commitState(snapshot.getId());
-    } catch (CoordinatorException e) {
-      TransactionState state = abort(id);
-      if (state.equals(TransactionState.ABORTED)) {
-        rollbackRecords(snapshot);
-        throw new CommitException(
-            "committing state in coordinator failed. the transaction is aborted", e);
+      Coordinator.State state = new Coordinator.State(id, TransactionState.COMMITTED);
+      coordinator.putState(state);
+      logger.debug(
+          "transaction {} is committed successfully at {}", id, System.currentTimeMillis());
+    } catch (CoordinatorConflictException e) {
+      try {
+        Optional<Coordinator.State> s = coordinator.getState(id);
+        if (s.isPresent()) {
+          TransactionState state = s.get().getState();
+          if (state.equals(TransactionState.ABORTED)) {
+            rollbackRecords(snapshot);
+            throw new CommitException(
+                "committing state in coordinator failed. the transaction is aborted", e);
+          }
+        } else {
+          throw new UnknownTransactionStatusException(
+              "committing state failed with NoMutationException but the coordinator status doesn't exist",
+              e,
+              id);
+        }
+      } catch (CoordinatorException e1) {
+        throw new UnknownTransactionStatusException("can't get the state", e1, id);
       }
+    } catch (CoordinatorException e) {
+      throw new UnknownTransactionStatusException("coordinator status is unknown", e, id);
```

Comment:
I think we can just throw `UnknownTransactionStatusException` when catching `CoordinatorException` because in that case, something unexpected database error is happening, so there is probably nothing we can do.